File: app/src/main/java/com/seafile/seadroid2/framework/model/ErrorModel.java
package com.seafile.seadroid2.framework.model;

import android.text.TextUtils;

import java.util.List;

public class ErrorModel {
    public String error_msg;
    public String msg;
    public String error;
    public String errorMsg;
    public String errMessage;
    public String err_message;
    public String detail;
    public List<String> non_field_errors;

    public String getError() {
        if (!TextUtils.isEmpty(error_msg)) return error_msg;
        if (!TextUtils.isEmpty(error)) return error;
        if (!TextUtils.isEmpty(errorMsg)) return errorMsg;
        if (!TextUtils.isEmpty(detail)) return detail;
        if (!TextUtils.isEmpty(errMessage)) return errMessage;
        if (!TextUtils.isEmpty(err_message)) return err_message;
        if (non_field_errors != null && !non_field_errors.isEmpty()) {
            return non_field_errors.get(0);
        }
        if (!TextUtils.isEmpty(msg)) return msg;
        return null;
    }
}

File: app/src/main/java/com/seafile/seadroid2/framework/model/ErrorModel.kt
package com.seafile.seadroid2.framework.model

import com.google.gson.annotations.SerializedName

class ErrorModel {
    @SerializedName("error_msg")
    var error_msg: String? = null
    var msg: String? = null

    @SerializedName("error")
    var errorRaw: String? = null

    var errorMsg: String? = null
    var errMessage: String? = null

    @SerializedName("err_message")
    var err_message: String? = null

    var detail: String? = null

    @SerializedName("non_field_errors")
    var non_field_errors: List<String>? = null

    fun getError(): String? {
        if (!error_msg.isNullOrEmpty()) return error_msg
        if (!errorRaw.isNullOrEmpty()) return errorRaw
        if (!errorMsg.isNullOrEmpty()) return errorMsg
        if (!detail.isNullOrEmpty()) return detail
        if (!errMessage.isNullOrEmpty()) return errMessage
        if (!err_message.isNullOrEmpty()) return err_message
        non_field_errors?.firstOrNull()?.let { return it }
        if (!msg.isNullOrEmpty()) return msg
        return null
    }
}
